nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -69,25 +84,27 @@ void Pooling2DLayer::forwarding(bool training) {
     hidden_ = Tensor(hidden_dim);
   }
 
-  if (pooling_type == PoolingType::max) {
-    max_idx.resize(output_dim[0].getDataLen());
-  } else if (pooling_type == PoolingType::global_max) {
-    max_idx_global.resize(output_dim[0].getDataLen());
+  if (training) {
+    if (pooling_type == PoolingType::global_max) {
+      max_idx_global.clear();
+    } else {
+      max_idx.clear();
+    }
   }
 
   for (unsigned int b = 0; b < in_dim.batch(); ++b) {
-    Tensor in_padded;
-    zero_pad(b, input_, padding.data(), in_padded);
+    Tensor in_sub = input_.getBatchSlice(b, 1);
     Tensor result = hidden_.getBatchSlice(b, 1);
-    result = pooling2d(b, in_padded, result);
+    // zero_pad(b, input_, padding.data(), in_padded);
+    pooling2d(in_sub, training, result);
   }
 }
 
 void Pooling2DLayer::calcDerivative() {
   unsigned int batch = input_dim[0].batch();
   unsigned int channel = input_dim[0].channel();
-  unsigned int height = input_dim[0].height();
-  unsigned int width = input_dim[0].width();
+  int height = input_dim[0].height();
```

Comment:
intended?

Reply:
Yes, this reduces calculation when for-looping

Reply:
For calcDerivative, I guess this can stay unsigned.

Reply:
Done

Reply:
Found that this is need for the next commit. For this commit, it is unsigned but for the next commit it becomes `int`